ocaml/lambda/transl_array_comprehension.ml

```diff
@@ -494,12 +495,13 @@ let iterator ~transl_exp ~scopes ~loc
              ; for_body   =
                  Matching.for_let
                    ~scopes
+                   ~arg_sort:Sort.for_predef_param
+                   ~return_layout:(Pvalue Pintval)
```

Comment:
Note that here I've just given a label to the pre-existing argument to this function.

As for why `Pintval` is the right value kind: I guess it's because here we're translating an iterator by building a for loop that applies the a function to each element of the array, and that function should return unit.

Reply:
Ah, yes, that's right. I got thrown off the scent as I was reading.